presto-native-execution/presto_cpp/main/operators/StorageBasedBroadcastWrite.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/StorageBasedBroadcastWrite.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace facebook::presto::operators {
+namespace {
+
+class StorageBasedBroadcastWrite : public Operator {
+ public:
+  StorageBasedBroadcastWrite(
+      int32_t operatorId,
+      DriverCtx* FOLLY_NONNULL ctx,
+      const std::shared_ptr<const StorageBasedBroadcastWriteNode>& planNode)
+      : Operator(
+            ctx,
+            planNode->outputType(),
+            operatorId,
+            planNode->id(),
+            "StorageBasedBroadcastWrite") {
+    // Assign a unique file name by combining driver and operator Ids.
+    auto filePath = fmt::format(
+        "{}/{}_{}.bin", planNode->resultFolder(), operatorId, ctx->driverId);
+    file_ = planNode->fileSystem()->openFileForWrite(filePath);
+    VELOX_CHECK(
+        file_ != nullptr, "Unable to open the broadcast file " + filePath + "!")
```

Comment:
Do not use ! in the error messages. No need to yell.

Use VELOX_CHECK_NOT_NULL with format arguments:

```
VELOX_CHECK_NOT_NULL(file, "Unable to open the broadcast file {}.", filePath);
```